### PR TITLE
[KATC] Set immutable option when opening sqlite db

### PR DESCRIPTION
**Pull request by RebeccaMahany**
Fixes the following:

> error generating table: fetching data: querying /path/to/some/db.sqlite: running query: database is locked (5) (SQLITE_BUSY)

### CODE REVIEW

File: ee/katc/sqlite.go [hunk 1]
```diff
@@ -118,3 +132,34 @@ func querySqliteDb(ctx context.Context, slogger *slog.Logger, path string, query
 
 	return results, nil
 }
+
+// copySqliteDb makes a temporary directory and copies the given db into it.
+func copySqliteDb(path string) (string, error) {
+	dbCopyDir, err := agent.MkdirTemp("sqlite-temp")
```
Comment (directionless) [inline ee/katc/sqlite.go:138]: If we have to, we have to... Did we explore the sqlite options that might alleviate this?

Comment (RebeccaMahany) [reply ee/katc/sqlite.go:138]: `mode=ro` wasn't making a difference -- let me try setting `immutable` to see!

Comment (RebeccaMahany) [reply ee/katc/sqlite.go]: It works! Thank you, that's much tidier

**Review by zackattack01 — APPROVED** [review]

File: ee/katc/sqlite.go (hunk 1 above)
Comment (directionless) [reply ee/katc/sqlite.go]: The docs say `database is opened read-only and all locking and change detection is disabled` so I was hopeful

**Pull request merged**